### PR TITLE
Add CoC Confirmation Decision to Route 11

### PR DESCRIPTION
**Pull request by dtgreiner**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description
Add a new decision to the end of route 11 that allows the CoC to confirm the success of the match.

Also included is a script that will clear out any route 11 matches (unless on production)

## Type of change
[//]: # 'remove options that are not relevant'
- [X] New feature (adds functionality)

## Checklist before requesting review
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [X] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [X] My code follows the style guidelines of this project (rubocop)
- [X] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: app/models/match_decisions/base.rb [hunk 1]
```diff
@@ -86,7 +86,7 @@ def contact_name
     def editable?
       # can this decision be updated by a notification response?
       # override this default behavior in subclasses
-      initialized? && match_open? && saved_status !~ /\A(accepted|confirmed|declined|canceled|rejected|complete|completed|scheduled|no_hearing|mitigation_required|mitigation_not_required|decline_overridden|decline_overridden_returned|decline_confirmed)\z/
+      initialized? && match_open? && saved_status !~ /\A(accepted|acknowledged|confirmed|declined|canceled|rejected|complete|completed|scheduled|no_hearing|mitigation_required|mitigation_not_required|decline_overridden|decline_overridden_returned|decline_confirmed)\z/
```
Comment (eanders) [inline app/models/match_decisions/base.rb]: @dtgreiner why this change?  I feel like we removed `acknowledged` because it was causing problems once deployed.

File: app/models/match_decisions/eleven/eleven_confirm_match_success_dnd_staff.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+#
+# License detail: https://github.com/greenriver/boston-cas/blob/production/LICENSE.md
+###
+
+module MatchDecisions::Eleven
+  class ElevenConfirmMatchSuccessDndStaff < ::MatchDecisions::Base
+    include MatchDecisions::AcceptsDeclineReason
+    include MatchDecisions::RouteElevenDeclineReasons
+    include MatchDecisions::RouteElevenCancelReasons
+    # validate :note_present_if_status_rejected
```
Comment (eanders) [inline app/models/match_decisions/eleven/eleven_confirm_match_success_dnd_staff.rb]:
```suggestion
```

File: app/models/match_decisions/eleven/eleven_confirm_match_success_dnd_staff.rb [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    def to_partial_path
+      'match_decisions/eleven/confirm_match_success_dnd_staff'
+    end
+
+    def statuses
+      {
+        pending: 'Pending',
+        confirmed: 'Confirmed',
+        rejected: 'Rejected', # Not currently used
```
Comment (eanders) [inline app/models/match_decisions/eleven/eleven_confirm_match_success_dnd_staff.rb]: If this isn't used:
```suggestion
```

File: db/migrate/20241010194153_destroy_route_eleven_matches.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class DestroyRouteElevenMatches < ActiveRecord::Migration[7.0]
+  def change
```
Comment (eanders) [inline db/migrate/20241010194153_destroy_route_eleven_matches.rb]:
```suggestion
  def up
```

File: app/models/match_decisions/base.rb (hunk 1 above)
Comment (dtgreiner) [reply app/models/match_decisions/base.rb]: Thats right, i do remember that one. I added it here because of a display bug after the first step was acknowledged. It wasn't showing the completed checkmark in the breadcrumbs.
Screenshot 2024-10-11 at 8 15 30 AM

I'll pull it out of here as this is less disruptive than the problems it was causing earlier. I can create a ticket to find a good resolution for the two issues.

File: db/migrate/20241010194153_destroy_route_eleven_matches.rb (hunk 4 above)
Comment (dtgreiner) [reply db/migrate/20241010194153_destroy_route_eleven_matches.rb]: This has been added in the latest commit

File: app/models/match_decisions/eleven/eleven_confirm_match_success_dnd_staff.rb (hunk 3 above)
Comment (dtgreiner) [reply app/models/match_decisions/eleven/eleven_confirm_match_success_dnd_staff.rb]: This has been included in the latest commit

File: app/models/match_decisions/eleven/eleven_confirm_match_success_dnd_staff.rb (hunk 2 above)
Comment (dtgreiner) [reply app/models/match_decisions/eleven/eleven_confirm_match_success_dnd_staff.rb]: This has been included in the latest commit

File: app/models/match_decisions/base.rb (hunk 1 above)
Comment (dtgreiner) [reply app/models/match_decisions/base.rb]: [Ticket added here](https://github.com/open-path/Green-River/issues/6835)

**Comment by dtgreiner** [comment]
Cleaned up the current decision info display:

Successful matches will show as such
Screenshot 2024-10-11 at 11 39 57 AM

If the current decision is nil, [_restrictions_on_reopening.haml](https://github.com/greenriver/boston-cas/blob/dg/housing_nav_route_confirm-6777/app/views/matches/_restrictions_on_reopening.haml#L1) is displayed (also checks if `@match.can_be_reopened?`
Screenshot 2024-10-11 at 11 39 49 AM

Match in progress shows as 
Screenshot 2024-10-11 at 11 49 48 AM

File: app/views/matches/_current_step_info.haml [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
-              = match.overall_status[:name]
-            -else
-              = @match.current_decision.step_name
-        .summary-readout--tile
-          %h3.h4 Assigned To
-          %h1.h2
-            = @match.current_decision.try(:actor_type) || 'N/A'
-      = render 'match_list_base/current_step', match: @match
-  - else
+  - unless @match.current_decision.present?
```
Comment (eanders) [inline app/views/matches/_current_step_info.haml]: So much better, thank you!
one tiny tweak:
```suggestion
  - if @match.current_decision.blank?
```
